src/fetchers.ts

```diff
@@ -115,9 +115,30 @@ export interface VersionResult {
   versions?: Version[];
 }
 
-export interface Version {
-  main: string;
-  app?: string;
+export class Version {
+  public readonly semantic: semver.SemVer | null;
```

Comment:
Do we have an example case where this might be null? Having this possibly null means the ordering might be surprising. I'm tempted to say that if we can't get semver to recognize something then we should drop the version and log a warning.

Reply:
It seems the only way to get this to be `null` from `semver.coerce` is if the version string does not have any numbers inside it whatsoever. Theoretically you could have a version be "I have 2 apples" and it would be coerced to `2.0.0`. I'm actually a bit disappointed with `coerce` since it doesn't bother with prerelease versions either so if your string looks like `myversion-1.0.0-alpha.1` it gets coerced to just `1.0.0`.

Reply:
I just made it throw an error if the provided version couldn't be coerced. It was simpler to do it this way rather than filter + warn.

The main place this is an issue is with the `html` fetcher since we're pretty much grabbing random elements and hoping for the best.

Reply:
I just wanted to avoid invalid data in the app if we could. Filtering or throwing are equally good at that. 👍